### ISSUE TITLE
Padding Token Bug in SDXL CLIP-G Text Encoder Causing Noisy "!" Tokens

### ISSUE THREAD

**Issue by terracottahaniwa** [comfyanonymous/ComfyUI#9849]
### Custom Node Testing

- [x] I have tried disabling custom nodes and the issue persists (see [how to disable custom nodes](https://docs.comfy.org/troubleshooting/custom-node-issues#step-1%3A-test-with-all-custom-nodes-disabled) if you need help)

### Expected Behavior

CLIP-G text encoder correctly pads prompts using the <|endoftext|> token without introducing noisy "!" tokens.

### Actual Behavior

Prompts shorter than 75 tokens are padded with "!" tokens internally, which act as noise and degrade image generation quality.

### Steps to Reproduce

Use ComfyUI with SDXL and input a prompt shorter than 75 tokens; although tokens are not displayed in the console, the internal padding can be inferred from debugging or testing different prompts.

### Debug Logs

```powershell
Tokenization details are not output in the console. The issue was detected via source code inspection and observed degradation in generated images linked to padding behavior.
```

### Other

ComfyUI v0.3.59

There is a similar report already opened in #9844, but it is reasonable to consider this as a likely bug, so I am posting this. However, it is also possible that the current implementation is intentional and the [original report (Japanese)](https://note.com/gentle_murre488/n/n12f2ecce1e00) may be incorrect. 
This issue broadly affects workflows including HiDream.

**Description:**  
When using the SDXL model in ComfyUI, there appears to be an issue with how the CLIP-G text encoder handles padding for input prompts shorter than 75 tokens. Instead of padding with the correct end-of-text token, the encoder fills the remaining token space with "!" tokens. These tokens are then disproportionately prioritized during attention, effectively acting as noisy pseudo-keywords that can degrade image generation quality.

The problem was traced to two root causes:  
1. Attention masks were disabled in the SDXLClipG class, causing the model to attend to padding tokens.  
2. The tokenizer used an incorrect padding setting (`pad_with_end=False`) and did not specify the correct pad token ID.

A fix involves:  
- Enabling attention masks in the SDXLClipG encoder (`enable_attention_masks=True`) to ignore padding tokens.  
- Removing `pad_with_end=False` and explicitly setting the pad token to the `<|endoftext|>` token (ID 49407) in the SDXLClipGTokenizer.  

After these changes, the noisy "!" tokens no longer appear, and padding behaves consistently with CLIP-L handling. This suggests the current unpatched behavior may introduce unintended noise and should be considered a bug.

The original report publishes the modified `sdxl_clip.py`.

Original report (Japanese):
https://note.com/gentle_murre488/n/n12f2ecce1e00

**Comment by comfyanonymous** [comment]
This is normal, the openclip models were trained with padding tokens of zero and SDXL was trained with zero padding tokens for the clip g with no attention masks on the clip.
